Dependencies/xr/Source/ARKit/XR.mm

```diff
@@ -1078,6 +1303,35 @@ void UpdatePlane(std::vector<Frame::Plane::Identifier>& updatedPlanes, Frame::Pl
             updatedPlanes.push_back(plane.ID);
         }
 
+        void UpdateMesh(std::vector<Frame::Mesh::Identifier>& updatedMeshes, Frame::Mesh& mesh, std::vector<Vector3f>& vertexBuffer, std::vector<uint32_t>& indexBuffer, std::vector<Vector3f>& normalsBuffer) {
+            // Check if mesh was actually updated
+            BOOL checkIfMeshUpdated = false;
+            if(mesh.Positions.size() != vertexBuffer.size()) {
+                checkIfMeshUpdated = true;
+            } else {
+                for (size_t i = 0; i < vertexBuffer.size(); i++) {
+                    if (abs(mesh.Positions[i].X - vertexBuffer[i].X) > 0.02f
```

Comment:
it seems like this check might be more expensive than the below copy operation. Could we use something like memcmp on the data vectors instead?